lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -289,14 +293,26 @@ def _monitor(self, poll_interval):
         Main thread that sends a stats request at the configured interval in
         seconds.
         """
+        while not self.init_finished:
+            # Still send an empty report -> for pipelined setup
+            self._report_usage({})
+            hub.sleep(self.INIT_SLEEP_TIME)
         while True:
-            for _, datapath in self.dpset.get_all():
-                if self.init_finished:
-                    self._poll_stats(datapath)
-                else:
-                    # Still send an empty report -> needed for pipelined setup
-                    self._report_usage({})
             hub.sleep(poll_interval)
+            for _, datapath in self.dpset.get_all():
+                now = datetime.now()
+                # If we're struggling to report give it some time to recover
+                if self._last_poll_time > self._last_report_timestamp:
+                    self.logger.error('Previous update running late, sleeping')
+                    self._last_report_timestamp = now
```

Comment:
Because this is an edge where we didn't get any reports for a while and we try to fix that by sleeping 2times. updating the report timestamp in case that one never gets updated so we don't get stuck forever

Reply:
Basically, if we don't update the timestamp and for some reason we never reach the future callback we would never poll again